docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst

```diff
@@ -0,0 +1,174 @@
+Squeeze
+=======
+
+
+.. meta::
+  :description: Learn about Squeeze-15 - a shape manipulation operation, which
+                can be performed on one required and one optional input tensor.
+
+**Versioned name**: *Squeeze-15*
+
+**Category**: *Shape manipulation*
+
+**Short description**: *Squeeze* removes dimensions equal to 1 from the first input tensor.
+
+**Detailed description**: *Squeeze* can be used with or without the second input tensor.
+
+* If only the first input is provided, every dimension that is equal to 1 will be removed from it.
+* With the second input provided, each value is an index of a dimension from the first tensor that is to be removed. Specified dimension should be equal to 1, otherwise it will be ignored and copied as is.
+  Dimension indices can be specified directly, or by negative indices (counting dimensions from the end).
+
+.. note::
+
+    - If index of the dimension to squeeze is provided as a constant input and it points to a dynamic dimension that might be `1`, then the dimension is considered as squeezable. Therefore the rank of the output shape will be reduced, but not dynamic.
+    - If the input with indices is empty or not provided, dynamic dimension compatible with `1` leads to dynamic rank of the output shape.
+
+
+**Attributes**:
+
+* *allow_axis_skip*
+
```

Comment:
Do we expect exception raised when `allow_axis_skip=False` and given axis corresponds to dimension that is not squeezable?

Reply:
It could be implemented to throw, but the intention is to keep the v0::Squeeze compatible behavior when `allow_axis_skip=False`.
For model enablement the v0::Squeeze has been already updated to be not so strict for static dimensions, but it was not possible to update the all dynamic cases and keep backward compatibility. Especially previous Squeeze logic assumes that dynamic dim with 1 in range like {-1} or {1-2} is squeezable, and produces static rank. 
While the recent request to align with torch is to produce dynamic rank for such case.
In fact `allow_axis_skip=False` can be understood like `assume_dynamic_dim_is_squeezable=True`.

Reply:
It looks that this mode is also helpful for TF Squeeze op (https://www.tensorflow.org/api_docs/python/tf/raw_ops/Squeeze) when axis = []. For such case, I can compute axes for our Squeeze using `ShapeOf` and `Range` ops and set this mode `allow_axis_skip=False`. This way it will squeeze all dims that are possible to squeeze. So we can squeeze dimension during both `conversion/loading` and `inference`, right?